### PR TITLE
Add `no-duplicate-super`

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [X] Addresses an existing issue: #1983
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update
- [X] Enable CircleCI for your fork (https://circleci.com/add-projects)

#### What changes did you make?

Added the `no-duplicate-super` rule, which warns if `super()` appears twice in a constructor.

### CODE REVIEW

File: test/rules/no-duplicate-super/test.ts.lint [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+class D {
+    // Unfortunately, we warn here too.
+    constructor() {
+        if (cond()) {
+            super(true);
+        } else {
+            super(false);
+            ~~~~~ [0]
```
Comment (ajafff) [inline test/rules/no-duplicate-super/test.ts.lint]: I think this one should be doable. This could result in many false positives in real world code.

The following would be more challenging and I don't know if we want to support code like that:
```typescript
constructor() {
  if (cond)
    super(true);
  if (!cond)
    super(false);
}
```

And then there is this nasty but totally valid constructor:
```typescript
constructor() {
  if(cond)
    return super(foo);
  if(someOtherCondition)
    return super(bar);
  return super(baz);

Comment (andy-hanson) [inline test/rules/no-duplicate-super/test.ts.lint]: What if the `if` has a `return` instead of `else`? What if there's a `switch` statement with `break`?
TypeScript does control flow analysis, but as far as I can tell, there's no API for accessing it. But I would hate to add a bunch of exceptions that duplicate work done better by TypeScript itself.

File: test/rules/no-duplicate-super/test.ts.lint [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        }
+    }
+}
+
+[0]: 'super()' is used twice. It must be called only once.
```
Comment (ajafff) [inline test/rules/no-duplicate-super/test.ts.lint]: would reword to something like: `multiple calls to super() found. It must be called only once.`

**Comment by nchen63** [comment]
I don't think this should go in until it can use control flow analysis. There's going to be a lot of false positives.

**Comment by andy-hanson** [comment]
Added some minimal control flow analysis. Tell me if there are any test cases I missed.

**Comment by andy-hanson** [comment]
Not sure why the tests are failing. It only happens in one of the 4 containers.

**Comment by nchen63** [comment]
Container 3 tests against typescript 2.0.10. I don't know why there aren't any failures in the log, though.

**Comment by andy-hanson** [comment]
I rebased and it works now.
